### PR TITLE
ForUs authentication module

### PR DESCRIPTION
**Pull request by OmarIthawi**
**Tasks:**

 - [ForUs Auth cryptography module](https://app.asana.com/0/23394127591897/78439276700871)
 - [Display a minimal registration form and request the students to sign the code of honor](https://app.asana.com/0/23394127591897/79652586630989)
 - [Disallow account merges](https://app.asana.com/0/23394127591897/78439276700879)

**PRs:**

 - Platfrom: https://github.com/Edraak/edx-platform/pull/144
 - Configuration: https://bitbucket.org/Edraak/configuration/pull-requests/21
**Tasks:**

 - [ForUs Auth cryptography module](https://app.asana.com/0/23394127591897/78439276700871)
 - [Display a minimal registration form and request the students to sign the code of honor](https://app.asana.com/0/23394127591897/79652586630989)
 - [Disallow account merges](https://app.asana.com/0/23394127591897/78439276700879)

**PRs:**

 - Platfrom: https://github.com/Edraak/edx-platform/pull/144
 - Configuration: https://bitbucket.org/Edraak/configuration/pull-requests/21/wip-forus-configs

### CODE REVIEW

**Comment by devalih** [comment]
Security wise looks good, bit I would limit the access to POST only.

There's some cases that are missing, I'll put it on Asana

File: lms/djangoapps/edraak_forus/views.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+class UpdateLanguageView(View):
+    def get(self, request):
+        self.update_language()
+        return redirect(reverse('forus_v1_auth_validate') + '?' + request.META['QUERY_STRING'])
+
+    def update_language(self):
+        url_lang = self.request.GET.get('lang')
+        if check_for_language(url_lang):
+            self.request.session[LANGUAGE_SESSION_KEY] = url_lang
```
Comment (devalih) [inline lms/djangoapps/edraak_forus/views.py]: I don't like this, but I can't think of a better solution for now.

please at a todo to refactor this.

Comment (OmarIthawi) [inline lms/djangoapps/edraak_forus/views.py]: It is simple, stupid and it works. Why don't you like it?

Comment (devalih) [inline lms/djangoapps/edraak_forus/views.py]: Or you know what, change language should be done in a middleware.

Comment (OmarIthawi) [inline lms/djangoapps/edraak_forus/views.py]: 😁

On Tue, Feb 2, 2016 at 10:58 AM, Ali <notifications@github.com> wrote:

> In lms/djangoapps/edraak_forus/views.py
> <https://github.com/Edraak/edx-platform/pull/144#discussion_r51540064>:
>
> > +
> > +from .models import ForusProfile
> > +from .helpers import validate_forus_params, forus_error_redirect
> > +
> > +log = logging.getLogger(__name__)
> > +
> > +
> > +class UpdateLanguageView(View):
> > +    def get(self, request):
> > +        self.update_language()
> > +        return redirect(reverse('forus_v1_auth_validate') + '?' + request.META['QUERY_STRING'])
> > +
> > +    def update_language(self):
> > +        url_lang = self.request.GET.get('lang')
> > +        if check_for_language(url_lang):
> > +            self.request.session[LANGUAGE_SESSION_KEY] = url_lang
>
> Or you know what, change language should be done in a middleware.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Edraak/edx-platform/pull/144/files#r51540064>.
>



-- 
Omar Al-Ithawi

Senior Engineer, Edraak.org <http://www.edraak.org/>


Queen Rania Foundation

for Education & Development

T +962 6 4016464  Ext. 700

M +962 7 90574405

E oithawi@qrf.org

<https://twitter.com/edraak> <https://www.facebook.com/edraakorg>
<http://www.youtube.com/user/edraakorg> <http://instagram.com/edraakorg>

**Comment by devalih** [comment]
Looks good, merging.

**Comment by devalih** [comment]
Or you know what, just squash the commits first 

WIP WIP WIP WIP WIP
Hotfix Hotfix Hotfix Hotfix Hotfix
Fixup Fixup Fixup Fixup Fixup

**Comment by OmarIthawi** [comment]
@devalih Done

**Pull request merged**